opm/grid/GraphOfGrid.cpp

```diff
@@ -279,17 +282,20 @@ void GraphOfGrid<Grid>::mergeWellIndices(const std::set<int>& well)
         if (newWell.find(idx) != newWell.end()) {
             continue;
         }
-        for (auto w = wells.begin(); w != wells.end(); ++w) {
+        for (auto w = wells.begin(); w != wells.end();) {
             if (w->find(idx) != w->end()) {
                 // idx is in another well => remap it and join wells
                 if ( wellIdx == idx ) {
                     wellIdx = *(w->begin());
                 }
                 idx = *(w->begin());
                 newWell.insert(w->begin(), w->end());
-                wells.erase(w);
+                w = wells.erase(w);
```

Comment:
unnecessary assignment as you're immediately breaking

Reply:
done!